controllers/articles.controller.js

```diff
@@ -15,6 +17,28 @@ exports.getArticleById = (req, res, next) => {
             res.status(200).send({ article });
         })
         .catch((err) => {
-            next(err);
+            if (err.code) {
+                psqlErrorHandler(err, next);
+            } else {
+                next(err);
+            }
+        });
+};
+
+exports.getCommentsByArticleId = (req, res, next) => {
+    const article_id = req.params.article_id;
+    return selectArticleById(article_id, next)
+        .then(() => {
+            return selectCommentsByArticleId(article_id);
+        })
+        .then((comments) => {
+            res.status(200).send({ comments });
+        })
+        .catch((err) => {
```

Comment:
You shouldn't need if/else logic here. The error will be passed through the error handling functions through next, with the function being responsible for controlling if an error is relevant or not